src/lib.rs

```diff
@@ -264,6 +266,77 @@ async fn create_habit(
     Ok(HttpResponse::Created().finish())
 }
 
+#[delete("/habit")]
+async fn delete_habit(
+    req: HttpRequest,
+    habit: web::Json<Habit>,
+    state: web::Data<State>,
+) -> Result<HttpResponse, CustomError> {
+    info!("Delete habit");
+    let username = authenticate(req.headers()).map_err(CustomError::AuthError)?;
+
+    let mut stmt_result = state.conn.lock().expect("failed to lock conn");
+    let conn = &mut *stmt_result;
+    let tx = conn.transaction().unwrap();
+
+    let result: Option<i64> = tx
```

Comment:
Could you give a more suggesting name to this variable?
Like habitId

Reply:
done